dashboard/dashapi/dashapi.go

```diff
@@ -40,9 +40,42 @@ func New(client, addr, key string) (*Dashboard, error) {
 	return NewCustom(client, addr, key, http.NewRequest, http.DefaultClient.Do, nil, nil)
 }
 
-func NewMock(mocker RequestMocker) *Dashboard {
+// MockControl can be used to dynamically mock dashboard methods.
+// Go does not let us define per-method template parameters,
+// so there's a separate SetHandler() method to add new mocks.
+type DashboardMock struct {
```

Comment:
proxyappclient_test.go has the usage example

Reply:
It looks like, in order to use `testify.Mock`, I'd need to:
a) Extract an interface out of `dashapi.Dashboard` (currently it's a struct).
b) Define stubs for all `dashapi.Dashboard` methods in the test file -- and do it in every package where I'd use the mock, since `_test.go` are not really exportable.

Or at least only do it for the methods used in `syz-ci`, sth like `syz-ci`'s view of the dashboard API (and therefore the interface will be `syz-ci`-only).

Only after that I will be able to pass the mocked dashboard object as if it were the original.

I hoped the library would deprive me of the need to do all that, but it's only simplifying the `if argument is X, return Y` part, which is very easy here anyway.

Reply:
These changes are not necessarily bad, it's probably even the opposite -- defining a separate interface with a subset of required API interfaces in every package that needs dashapi might make our interfaces only cleaner. But we need to understand whether it's worth doing now.